components/emoji_picker/modules/emoji_skin_selector.vue

```diff
@@ -0,0 +1,16 @@
+<template>
+  <div class="d-emoji-picker__skin-selector">
+    <!--    Skin Selector -->
+  </div>
+</template>
+
+<script setup>
+const emits = defineEmits([
+  /**
+   * The skin tone that was selected
+   * @event skin-tone
+   * @type {Number}
+   */
+  'skin-tone',
+]);
+</script>
```

Comment:
Can you comment a few more details on how this one is going to work?

Does it change the look of all the "skin tone based" emojis in the selector?
How does it work if you interact with it via keyboard?
left / right arrows to select skin tone?
Is there a way to access it without tabbing through the entire list of emojis?
Is there any other way to change skin tone when you are selecting an emoji?

Reply:
Sure.... this code is deprecated, because skin tone its not a number, its a composition CODE + LETTER.

<img width="364" alt="image" src="https://user-images.githubusercontent.com/89984179/226670429-0cbb6050-7684-4d42-99ab-e4c201bae451.png">

<img width="452" alt="image" src="https://user-images.githubusercontent.com/89984179/226670528-2266eeec-8be0-4b3c-bf60-1021dd4c548f.png">

For each mutable skin emoji we have 6 emojis

1. palms up together - 1f64c
2. palms up together: light skin tone - 1f64c-1f3fb
3. palms up together: medium-light skin tone - 1f64c-1f3fc
4. palms up together: medium skin tone - 1f64c-1f3fd
5. palms up together: medium-dark skin tone - 1f64c-1f3fe
6. palms up together: dark skin tone - 1f64c-1f3ff

Each skin tone its a composition 
{EMOJI_UNICODE}-{SKIN_CODE}{SKIN_TONE}
Emoji unicode: 1f64c
Skin code (SAME FOR ALL): 1f3f
Skin tone: b/c/d/e/f

So.. I will update this code to handle this behaviour.

![image](https://user-images.githubusercontent.com/89984179/226673676-feb70706-e05f-48c9-87a1-6f61bda9188d.png)

Reply:
This is excellent detail and exactly the type of thing developers would like to see here. Make sure to comment this into your component or include in documentation.

Reply:
Now knowing this, let me answer your questions:

- Does it change the look of all the "skin tone based" emojis in the selector?
Yes, as thats the slack behaviour, the only application I can remember now which has specific skin tone in each emoji its whatsapp

Slack skin tone for all emojis
<img width="357" alt="image" src="https://user-images.githubusercontent.com/89984179/226675160-b3a8888c-a3ad-4ece-84e9-7bf8b7109995.png">


Whatsapp skin tone for each emoji behaviour
<img width="599" alt="image" src="https://user-images.githubusercontent.com/89984179/226674973-3e2904f3-e1ac-4376-83da-f03a36861dcb.png">


- How does it work if you interact with it via keyboard?
There is no keyboard navigation specification.
In this case I would replicate the slack emoji picker navigation.

Its work in this way:

1. On emoji picker open it will focus the search input
2. After tab it will jump to the first icon of each tabset.
3. After navigate all the tabset its jump to an specific custom behaviour of "Add Item" (we dont have it) so we will jump directly to the skin tone selector.
4. After tab it will jump to the first Tab of the tabset.
5. After tab it will jump again to the search input.

When the focus is on the emoji selector, you can start to navigate through all the emojis with arrow keys and select with Enter.
When the focus is on the Tabset, you can start to navigate through all the Tabs with arrow keys and select with Enter.
When the focus is on the skin-tone selector you can "Enter" and open the skin selector, then you can navigate through all the skins with arrow keys and select with Enter.


- left / right arrows to select skin tone?
Yes

- Is there a way to access it without tabbing through the entire list of emojis?
I cannot found the shortcode to jump directly to skin tone on Slack behaviour.

- Is there any other way to change skin tone when you are selecting an emoji?
I cannot found the shortcode to to change skin tone on Slack behaviour.
On Whatsapp you can maintain click for 1 second and it will appear the skin selector above.

Reply:
> * Is there a way to access it without tabbing through the entire list of emojis?
>   I cannot found the shortcode to jump directly to skin tone on Slack behaviour.

I think you answered this by saying the emojis are navigated with arrow keys though, that is good enough!


> * Is there any other way to change skin tone when you are selecting an emoji?
>   I cannot found the shortcode to to change skin tone on Slack behaviour.
>   On Whatsapp you can maintain click for 1 second and it will appear the skin selector above.

Interesting, yeah don't think we need to worry about that here. Just the bottom skin selector should be fine.

Reply:
One more thing I realised doing this research.

We also have GENDER distinctions.
<img width="569" alt="image" src="https://user-images.githubusercontent.com/89984179/226683883-9ea00bac-fa72-489a-901b-b6ae8e657c3a.png">


And gender with skin tone distinctions.
<img width="569" alt="image" src="https://user-images.githubusercontent.com/89984179/226684024-a77aa7eb-5d37-4eae-b08b-2df0324b7677.png">


And some like this
<img width="620" alt="image" src="https://user-images.githubusercontent.com/89984179/226683674-48e467eb-8a32-4b4e-97d4-b5ad03eabacb.png">

Reply:
Yep and it also gets really crazy when there are multiple people in an emoji with different skin tones for example. Not sure what the plan is on these ones...